plugins/inputs/logfile/tail/tail.go

```diff
@@ -329,12 +332,14 @@ func (tail *Tail) tailFileSync() {
 		tail.Logger.Debugf("Seeked %s - %+v\n", tail.Filename, tail.Location)
 		if err != nil {
 			tail.Killf("Seek error on %s: %s", tail.Filename, err)
+			tail.TailerQueue.Dequeue()
```

Comment:
There are a lot of places where this `Dequeue()` call gets added. I think it makes more sense to trace through and identify where all of these different code paths converge and then only add one place where you dequeue. Also back to my other issue, you're not actually using any value from the queue (whether it is true or false). This is really just an atomic counter / counting semaphore, so if we're going to include this in some capacity, I would expect the naming to change.

Reply:
Yes. That's true. I have skimmed/trace through; however, I have not able to identify the `one place` to dequeue, that's why I have to spread out and deliver a solving solution and keep up the deadline.

Reply:
Resolve the issue by adding `tail.Deuque()` with defer (forgot about that): https://github.com/aws/amazon-cloudwatch-agent/pull/491/files#diff-b4cbda30ad21153305e4658dca8a5844580e52b12544b5147237a3315d185284R314